File: Problems/Validate Binary Search Tree/Mine.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int findLargest (TreeNode *root){
        TreeNode *previous = root;
        while (root != NULL){
            previous = root;
            root = root -> right;       
        }
        return previous -> val;
    }
    int findSmallest (TreeNode* root){
        TreeNode *previous = root;
        while (root != NULL){
            previous = root;
            root = root -> left;       
        }
        return previous -> val;
    }
    bool isValidBST(TreeNode* root) {
        if (root == NULL)
            return true;
        if (root -> left == NULL && root -> right == NULL)
            return true;
        if (root -> left == NULL && root -> right != NULL){
            if (isValidBST (root -> right) && root -> val < findSmallest (root -> right))
                return true;
            return false;
        }
        if (root ->left != NULL && root -> right == NULL)
        {
            if (isValidBST (root -> left) && root -> val > findLargest (root -> left))
                return true;
            return false;
        }
        if (isValidBST (root -> left) && isValidBST (root -> right) && root -> val > findLargest (root -> left) && root -> val < findSmallest (root -> right))
            return true;
        return false;
    }
};

File: Problems/701. Insert into a Binary Search Tree/mine.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def insertIntoBST(self, root: TreeNode, val: int) -> TreeNode:
        if not root:
            return TreeNode(val)
        
        current = root
        prev = None
        while (current):
            prev = current
            if val < current.val:
                current = current.left
                if not current:
                    prev.left = TreeNode(val)
                    break
            else:
                current = current.right
                if not current:
                    prev.right = TreeNode(val)
                    break
        
        return root
